extensions/src/main/scala/water/fvec/H2OFrame.scala

```diff
@@ -0,0 +1,180 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one or more
+* contributor license agreements.  See the NOTICE file distributed with
+* this work for additional information regarding copyright ownership.
+* The ASF licenses this file to You under the Apache License, Version 2.0
+* (the "License"); you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package water.fvec
+
+import java.io.File
+import java.net.URI
+
+import water._
+import water.parser.DefaultParserProviders.GUESS_INFO
+import water.parser.ParseSetup
+import water.parser.ParseSetup._
+import water.util.FrameUtils
+
+/**
+ * Wrapper around Java H2O Frame
+ *
+ * @param frameKey reference of new frame
+ * @param names    column names for new frame
+ * @param vecs     vectors composing new frame
+ */
+class H2OFrame private(frameKey: Key[Frame], names: Array[String], vecs: Array[Vec])
+  extends Frame(frameKey, names, vecs) with FrameOps {
+
+  /** Create a new H2OFrame based on existing Java Frame.
+   *
+   * Simple field copy, so the Frames share
+   * underlying arrays.  Recommended that the input Java Frame be dead after
+   * this call.
+   *
+   * @param fr Java frame
+   * @return new H2O frame with parsed data
+   */
+  def this(fr: Frame) = this(if (fr._key != null) fr._key else Key.make("dframe" + Key.rand()).asInstanceOf[Key[Frame]], fr._names, fr.vecs())
```

Comment:
nit: Can we split this expression into more lines?

Reply:
I think it won't be possible as it is in the constructor and `this(` call needs to be first in the method definiton

Reply:
But I tried to put each argument on a new line for better readability